### PR TITLE
feat(CX-3635):update my collection sticky tab 

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [CX-3635] 

### Description

- Update my collection sticky tab 

_At a next ticket, we will need to create some context to tell which tab is open - will leave that for a futher ticket_

https://user-images.githubusercontent.com/11945712/234909997-e0dd276c-e6db-4fa9-bd6c-ac5c65bc4132.mov

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- update my collection sticky tab - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3635]: https://artsyproduct.atlassian.net/browse/CX-3635?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by lordkiz — APPROVED** [review]
Looks v. good. Nice animations!

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+const Messages = ({
+  hasMarketSignals,
+  showNewWorksMessage,
+  showSubmissionMessage,
+}: {
+  hasMarketSignals: boolean
+  showNewWorksMessage: boolean
+  showSubmissionMessage: boolean
+}) => {
```
Comment (damassi) [inline src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:300]: #superminor - Any reason these aren't React.FCs?

File: src/app/utils/hooks/useMeasure.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  pageX: number
+  pageY: number
+}
+
+export const useMeasure = ({ ref }: { ref: MutableRefObject<any> }) => {
+  const [position, setPosition] = useState<null | Position>(null)
+
+  useEffect(() => {
+    if (ref.current) {
+      ;(ref.current as any).measure(
```
Comment (damassi) [inline src/app/utils/hooks/useMeasure.ts:17]: the `;(...)` is somewhat prone to breakage; can we assign this to a variable?

**Comment by damassi** [comment]
Animations look pretty good! I wonder if we might be able to shorten the x translation a bit to eliminate some of the opacity / outline overlap: 

Screenshot 2023-04-27 at 10 19 31 AM

A lot of the time with animations a slight nudge in a particular direction can trick the user into thinking that its going further than it seems, like an optical illusion; the trick is getting the timing right. But in this case, i think shortening the distance might be all that's needed. Otherwise, looks great. (And this is just IMO, feel free to ignore)

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:300]: no reason actually 😄 I will update it. Thank you

File: src/app/utils/hooks/useMeasure.ts (hunk 2 above)
Comment (MounirDhahri) [reply src/app/utils/hooks/useMeasure.ts:17]: Sure - will update

File: src/app/utils/hooks/useMeasure.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            pageX,
+            pageY,
+          })
+        },
+        () => {
+          console.error("measurement failed")
+        }
+      )
+    }
+  }, [JSON.stringify(position)])
```
Comment (olerichter00) [inline src/app/utils/hooks/useMeasure.ts:33]: Shouldn't `position` work here too? When we change the position, we create a new object with `setPosition` anyway.

```suggestion
  }, [position])
```

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+  showNewWorksMessage,
+  showSeparator,
+}) => {
+  const { showVisualClue } = useVisualClue()
+  const [selectedTab, setSelectedTab] = useState<CollectedTab>(null)
+
+  const showSubmissionMessage = showVisualClue("ArtworkSubmissionMessage")
+  const enableCollectedArtists = useFeatureFlag("AREnableMyCollectionCollectedArtists")
+
+  const showFilters = true
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:58]: 👀

**Review by olerichter00 — APPROVED** [review]
Great work 🌟

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx (hunk 4 above)
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:58]: I am removing it now after pushing the changes Chris reported

File: src/app/utils/hooks/useMeasure.ts (hunk 3 above)
Comment (MounirDhahri) [reply src/app/utils/hooks/useMeasure.ts:33]: that's a good comment - i should have probably explained further why I am tricking this effect. 
The main reason behind me serializing the json is to make sure we're applying a shallow comparison the position object here. In the video attached, you can see the output of the console.log from both approaches.



Uploading Screen Recording 2023-04-28 at 12.52.34.mov…

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx (hunk 4 above)
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:58]: this is here to prepare for the next ticket about the filters - but better to be removed maybe until I pick up the other ticket

File: src/app/utils/hooks/useMeasure.ts (hunk 3 above)
Comment (olerichter00) [reply src/app/utils/hooks/useMeasure.ts:33]: Thanks for the explanation 👍

**Comment by MounirDhahri** [comment]
@damassi here we go. adjusting the padding didn't work - what was possible thought without making the UX worse was to wait for the spill to get blue ( 200 ms - the default spring animation duration) then do the translate X

https://user-images.githubusercontent.com/11945712/235132886-db09ac58-d5e7-4031-b5f9-a8968d261b94.mov

File: src/app/utils/hooks/useMeasure.ts (hunk 3 above)
Comment (MounirDhahri) [reply src/app/utils/hooks/useMeasure.ts:31]: you're welcome of course 
sorry - the upload didn't work 

https://user-images.githubusercontent.com/11945712/235133362-a5c0b13b-f9c2-4e32-9af3-0c18d1ae2dfa.mov

Comment (olerichter00) [reply src/app/utils/hooks/useMeasure.ts:31]: Isn't there a risk to create an infinite loop when setting position inside of the effect?

Comment (MounirDhahri) [reply src/app/utils/hooks/useMeasure.ts:31]: yes, it's why I am making sure to compare the objects as a string. 
A better approach here would have been using a reference - issue is that the initial position always comes as 0. I can try it out

**Pull request merged**